sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1792,3 +1793,138 @@ def _df_set_column_unicode_key_impl(self, key, value):
+
+    Limitations
+    -----------
+    Reset the index of the DataFrame, and use the default one instead.
+    Parameters level, inplacem col_level, col_fill unsupported.
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/dataframe_reset_index_drop_False.py
+        :language: python
+        :lines: 36-
+        :caption: Reset the index of the DataFrame, and use the default one instead.
+                  The old index becomes the first column.
+        :name: ex_dataframe_reset_index
+
+    .. command-output:: python ./dataframe/dataframe_reset_index_drop_False.py
+        :cwd: ../../../examples
+
+    .. literalinclude:: ../../../examples/dataframe/dataframe_reset_index_drop_True.py
+        :language: python
+        :lines: 36-
+        :caption: Reset the index of the DataFrame, and use the default one instead.
+        :name: ex_dataframe_reset_index
+
+    .. command-output:: python ./dataframe/dataframe_reset_index_drop_True.py
+        :cwd: ../../../examples
+
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas DataFrame method :meth:`pandas.DataFrame.reset_index` implementation.
+
+   .. only:: developer
+
+       Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_reset_index*
+
+   Parameters
+   -----------
+   self: :class:`pandas.DataFrame`
+       input arg
+   level: :obj:`int`, `str`, `tuple`, or `list`, default None
+       *unsupported*
+   drop: :obj:`bool`, default False
+       Just reset the index, without inserting it as a column in the new DataFrame.
+   inplace: :obj:`bool`, default False
+       *unsupported*
+   col_level: :obj:`int`, `str`, default 0
+       *unsupported*
+   col_fill: :obj, default ''
+       *unsupported*
+
+   Returns
+   -------
+   :obj:`pandas.DataFrame`
+        DataFrame with the new index or None if inplace=True.
+   """
+
+    func_name = 'reset_index'
+
+    ty_checker = TypeChecker('Method {}().'.format(func_name))
+    ty_checker.check(self, DataFrameType)
+
+    if not (level is None or isinstance(level, types.Omitted)):
+        raise TypingError('{} Unsupported parameter level. Given: {}'.format(func_name, level))
+
+    if not (isinstance(drop, (types.Omitted, types.Boolean)) or drop is False):
+        ty_checker.raise_exc(drop, 'bool', 'drop')
+
+    if not (inplace is False or isinstance(inplace, types.Omitted)):
+        raise TypingError('{} Unsupported parameter inplace. Given: {}'.format(func_name, inplace))
+
+    if not (col_level == 0 or isinstance(col_level, types.Omitted)):
+        raise TypingError('{} Unsupported parameter col_level. Given: {}'.format(func_name, col_level))
+
+    if not (col_fill == '' or isinstance(col_fill, types.Omitted)):
+        raise TypingError('{} Unsupported parameter col_fill. Given: {}'.format(func_name, col_fill))
+
+    if not isinstance(drop, types.Literal):
```

Comment:
Below implementation will only work with Boolean literals, so the check must be aligned.